src/serde.rs

```diff
@@ -117,3 +142,204 @@ impl<'de> Deserializer<'de> {
+    type Error = SerdeConversionError;
+    fn try_from(item: serde_json::Value) -> ConvertResult<Self> {
+        use serde_json::Value;
+        Ok(match item {
+            Value::Null => BorrowedValue::Null,
+            Value::Bool(b) => BorrowedValue::Bool(b),
+            Value::Number(b) => {
+                if let Some(n) = b.as_i64() {
+                    BorrowedValue::I64(n)
+                } else if let Some(n) = b.as_u64() {
+                    if n > std::i64::MAX as u64 {
+                        return Err(SerdeConversionError::IntegerTooLarge);
+                    }
+                    BorrowedValue::I64(n as i64)
+                } else if let Some(n) = b.as_f64() {
+                    BorrowedValue::F64(n)
+                } else {
+                    return Err(SerdeConversionError::Oops);
+                }
+            }
+            Value::String(b) => BorrowedValue::String(b.into()),
+            Value::Array(a) => BorrowedValue::Array(
+                a.into_iter()
+                    .map(|v| v.try_into())
+                    .collect::<ConvertResult<Vec<BorrowedValue>>>()?,
+            ),
+            Value::Object(o) => BorrowedValue::Object(
+                o.into_iter()
+                    .map(|(k, v)| Ok((k.into(), v.try_into()?)))
+                    .collect::<ConvertResult<crate::value::borrowed::Map>>()?,
+            ),
+        })
+    }
+}
+
+impl<'value> TryInto<serde_json::Value> for BorrowedValue<'value> {
+    type Error = SerdeConversionError;
+    fn try_into(self) -> ConvertResult<serde_json::Value> {
+        use serde_json::Value;
+        Ok(match self {
+            BorrowedValue::Null => Value::Null,
+            BorrowedValue::Bool(b) => Value::Bool(b),
+            BorrowedValue::I64(n) => Value::Number(n.into()),
+            BorrowedValue::F64(n) => {
+                if let Some(n) = serde_json::Number::from_f64(n) {
+                    Value::Number(n)
+                } else {
+                    return Err(SerdeConversionError::NanOrInfinity);
+                }
+            }
+            BorrowedValue::String(b) => Value::String(b.to_string()),
+            BorrowedValue::Array(a) => Value::Array(
+                a.into_iter()
+                    .map(|v| v.try_into())
+                    .collect::<ConvertResult<Vec<Value>>>()?,
+            ),
+            BorrowedValue::Object(o) => Value::Object(
+                o.into_iter()
+                    .map(|(k, v)| Ok((k.to_string(), v.try_into()?)))
+                    .collect::<ConvertResult<serde_json::map::Map<String, Value>>>()?,
+            ),
+        })
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use crate::{json, BorrowedValue, OwnedValue};
+    use serde_json::{json as sjson, Value as SerdeValue};
+    use std::convert::TryInto;
+    #[test]
+    fn convert_owned_value() {
+        let v: OwnedValue = json!({
+            "int": 42,
+            "float": 7,
```

Comment:
curious, does a real float w/ decimal point not work because of round-trip inaccuracy?

Reply:
Nope that was just me being stupid :)

Reply:
good catch! 👍 fixed!